internal/directory/provider.go

```diff
@@ -68,101 +76,66 @@ func GetProvider(options Options) (provider Provider) {
 	switch options.Provider {
 	case auth0.Name:
 		serviceAccount, err := auth0.ParseServiceAccount(options)
-		if err == nil {
-			return auth0.New(
-				auth0.WithDomain(options.ProviderURL),
-				auth0.WithServiceAccount(serviceAccount))
+		if err != nil {
```

Comment:
So we now require the use of a directory provider? This is a regression in functionality.

Reply:
no I just rewrote the upper level function that if there's an error then `provider = nullProvider{}` and removed the defer() logic of setting values